src/lib/useDragAndZoom.tsx

```diff
@@ -249,14 +244,33 @@ export function useDragAndZoom(
 
             const factor = Math.pow(wheelZoomFactor, event.deltaY / 53);
 
-            const newViewPort: Bounds = [x - (x - temporaryViewport[0]) * factor, x + (temporaryViewport[1] - x) * factor]
+            const newViewPort: Bounds = [
+                x - (x - temporaryViewport[0]) * factor,
+                x + (temporaryViewport[1] - x) * factor,
+            ];
 
-            temporaryViewport = fitBoundsInMinVisibleX(newViewPort, temporaryViewport, options.xMinVisible)
+            temporaryViewport = fitBoundsInMinVisibleX(newViewPort, temporaryViewport, options.xMinVisible);
 
-            temporaryViewport = fitBoundsInLimit(
-                temporaryViewport,
-                options.boundsLimit,
-            );
+            temporaryViewport = fitBoundsInLimit(temporaryViewport, options.boundsLimit);
+
+            onEnd(temporaryViewport);
+        };
+
+        const onDoubleClick = (event: MouseEvent) => {
+            event.preventDefault();
+            event.stopPropagation();
+            const bounds = element.getBoundingClientRect();
+            const x = calcXInTemporaryViewport(event.pageX, bounds);
+
+            const factor = event.altKey ? 2 : 0.5;
+
+            const newViewPort: Bounds = [
+                x - (x - temporaryViewport[0]) * factor,
+                x + (temporaryViewport[1] - x) * factor,
+            ];
+
+            temporaryViewport = fitBoundsInMinVisibleX(newViewPort, temporaryViewport, options.xMinVisible);
+            temporaryViewport = fitBoundsInLimit(temporaryViewport, options.boundsLimit);
```

Comment:
Could you please extract the common part of onWheel and onDoubleClick to a function?